lib/editing/editor_comid.dart

```diff
@@ -0,0 +1,297 @@
+// Copyright (c) 2014, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library editor.codemirror;
+
+import 'dart:async';
+import 'dart:html' as html;
+
+import 'package:comid/codemirror.dart' hide Document;
+import 'package:comid/addon/mode/dart.dart';
+import 'package:comid/addon/mode/htmlmixed.dart';
+import 'package:comid/addon/edit/matchbrackets.dart';
+import 'package:comid/addon/edit/closebrackets.dart';
+import 'package:comid/addon/edit/show-hint.dart' as hints;
+import 'package:comid/addon/comment/comment.dart' as comments;
+
+import 'editor.dart' hide Position;
+import 'editor.dart' as ed show Position;
+
+export 'editor.dart';
+
+final CodeMirrorFactory codeMirrorFactory = new CodeMirrorFactory._();
+
+final _gutterId = 'CodeMirror-lint-markers';
+
+class CodeMirrorFactory extends EditorFactory {
+
+  CodeMirrorFactory._();
+
+  List<String> get modes => null; // TODO CodeMirror.MODES; unused
```

Comment:
If not, perhaps this wrapper around comid should have some hard-coded values? `dart, html, css`?